common/source/library/globus_args.c

```diff
@@ -91,7 +91,7 @@ globus_l_args_create_error_msg( char **        error_msg,
     usage_len = strlen( oneline_usage );
 
     len = strlen(p);
-    strncpy( &p[len], oneline_usage, usage_len );
+    memcpy( &p[len], oneline_usage, usage_len );
```

Comment:
p is created by malloc a few lines before (line 83), so it will not overlap with oneline_usage.
The problem with the original strncpy call is not any potential overlap between source and destination (like memcpy overlap is not allowed for strncpy either).
The problem is that this call deliberately uses a length one character short to, on purpose, not have room for the NULL byte at the end of the string, i.e. it is misused to act as memcpy. The compiler warns because strncpy is not intended to be used like this. The function to copy an exact number of bytes without adding a NULL at the end is memcpy, no strncpy.

Reply:
Now I understand it (also for my other question below).